baseapp/abci.go

```diff
@@ -266,6 +266,15 @@ func (app *BaseApp) CheckTx(req abci.RequestCheckTx) abci.ResponseCheckTx {
 // Regardless of tx execution outcome, the ResponseDeliverTx will contain relevant
 // gas execution context.
 func (app *BaseApp) DeliverTx(req abci.RequestDeliverTx) (res abci.ResponseDeliverTx) {
+	res = app.deliverTxWithoutEventHistory(req)
+	// When successful, remember event history.
+	if res.Code == sdkerrors.SuccessABCICode {
+		app.deliverState.eventHistory = append(app.deliverState.eventHistory, res.Events...)
+	}
+	return res
```

Comment:
What are your thoughts on better indicating our patches?
```suggestion
	// [AGORIC] Remember event history for successful deliveries.
	// deliverTxWithoutEventHistory is the upstream cosmos-sdk DeliverTx.
	res = app.deliverTxWithoutEventHistory(req)
	if res.Code == sdkerrors.SuccessABCICode {
		app.deliverState.eventHistory = append(app.deliverState.eventHistory, res.Events...)
	}
	return res
```

Reply:
I think comments on our inline contributions calling them out and explaining them is super helpful. Every time we upgrade we need to revisit these pieces of code and any hints, pointers, explainers help us.

Reply:
Done. Redundant with the github compare, extra maintenance, and we'll want to backfill on all the other changes, but it's probably worth it for nontrivial changes like this.